### PR TITLE
Implementation screen centring and dynamic super resolution 

### PR DESCRIPTION
**Pull request by alphanu1**
Screen centring has been implemented.

Implantation of dynamic width for super resolution, this is used to get as close to native resolution a possible while keeping the pixel clock high enough.
  
Added code to calculate best width.
Added 1 to the options for super resolution.

Issue, I feel I have missed something as this value shows null instead of 1 or dynamic. Could some one point me in the right direction to fix this please.

### CODE REVIEW

File: gfx/video_crt_switch.h [hunk 1]
```diff
@@ -33,6 +33,8 @@ void crt_aspect_ratio_switch(unsigned width, unsigned height);
 
 void crt_video_restore(void);
 
+static int crt_compute_dynamic_width(int width);
+
```
Comment (twinaphex) [inline gfx/video_crt_switch.h]: This doesn't make a lot of sense to me. Static function prototypes should not be in a header file.

Comment (alphanu1) [reply gfx/video_crt_switch.h]: Hmm this was changed. And is not static in my master

Comment (twinaphex) [reply gfx/video_crt_switch.h]: OK.

**Comment by twinaphex** [comment]
Also, try to use git pull --rebase in the future.

**Comment by twinaphex** [comment]
Another thing I'd like to see later is that we start moving the HAVE_VIDEOCORE functions outside of the video_crt_switch file so that we don't mix implementation-specific code with what should be fairly generic code for all platforms. I don't know where you would put this functionality instead, maybe it should be part of the display server interface, dunno. Just food for thought. I'll leave any specifics up to you.

**Comment by alphanu1** [comment]
@twinaphex yes I agree. It should be in display servers. This is one of the next things to look into.

Did you see my comment about the new number (1) I've added to the super resolution?

Quick question. How would we go around adding a video core dips_serv? If it's not too taxing I'll start implementing it

File: menu/menu_setting.c [hunk 2]
```diff
@@ -1300,8 +1300,10 @@ static int setting_uint_action_left_crt_switch_resolution_super(
       case 0:
          *setting->value.target.unsigned_integer = 3840;
          break;
-      case 1920:
+      case 1: /* for dynamic super resolution switching - best fit */
```
Comment (twinaphex) [inline menu/menu_setting.c]: This falls-through. I'm not sure if this is what you intended but it looks wrong at least. You need to probably insert a break; here.

**Pull request merged**